src/AuthenticationPortal.Web/Controllers/TokenAuthController.cs

```diff
@@ -0,0 +1,61 @@
+﻿using AuthenticationPortal.Contracts;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
+using System.Linq;
+
+
+namespace AuthenticationPortal.Web
+{
+    [Route("api/[controller]")]
+    public class TokenAuthController : Controller
+    {
+        private readonly ITokenAuthenticator tokenAuthentication;
+        public TokenAuthController(ITokenAuthenticator tokenAuthentication)
+        {
+            this.tokenAuthentication = tokenAuthentication;
+        }
+
+
+        // GET: api/<controller>
+        [HttpGet]
+        public IActionResult Get()
+        {
+            StringValues x;
+            if (Request.Headers.TryGetValue("AuthKey", out x))
+            {
+                Token token = new Token() { TokenString = x.First() }; ;
+                return tokenAuthentication.AuthenticateToken(token).Result;
+            }
+            else
+            {
+                throw new CustomException(803);
+            }
+
+        }
+
+        // GET api/<controller>/5
+        [HttpGet("{id}")]
+        public string Get(string token)
+        {
+            return "value";
+        }
+
+        // POST api/<controller>
```

Comment:
give proper names to methods

Reply:
Which method? Controller methods like get, put etc. or the Service class methods

Reply:
Done